### PR TITLE
Fix CVE-2023-39533: update msft-golang to use 1.19.12 which contains fix

### PR DESCRIPTION
**Pull request by bfjelds**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Update msft-golang to 1.19.12-1 to address https://nvd.nist.gov/vuln/detail/CVE-2023-39533.

msft-golang contains a submodule pointing at a commit in golang.  Updating the version of msft-golang indirectly updates the version of golang.

###### Change Log  
Update msft-golang to 1.19.12-1

###### Does this affect the toolchain?  
**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-39533

###### Test Methodology
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=409492&view=logs&j=7909073b-e471-53c1-2c44-627e2efc0564

### CODE REVIEW

File: SPECS/msft-golang/msft-golang.spec [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+Release:        1%{?dist}
+License:        BSD
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          System Environment/Security
+URL:            https://github.com/microsoft/go
+Source0:        https://github.com/microsoft/go/releases/download/v1.19.12-1/go.20230802.5.src.tar.gz
+Source1:        https://dl.google.com/go/go1.4-bootstrap-20171003.tar.gz
+Patch0:         go14_bootstrap_aarch64.patch
 %global goroot          %{_libdir}/golang
```
Comment (mfrw) [inline SPECS/msft-golang/msft-golang.spec:13]: I think even though the linter yells at us for these macros to be placed here, our convention generally has been to keep all the macros at the top of the spec file.

File: SPECS/msft-golang/msft-golang.spec [hunk 2]
```diff
@@ -1,3 +1,15 @@
+Summary:        Go
+Name:           msft-golang
+Version:        1.19.12
```
Comment (mfrw) [inline SPECS/msft-golang/msft-golang.spec:3]: Please run the buddy build before the merge to ensure the package builds in the pipeline as well.

File: SPECS/msft-golang/msft-golang.spec (hunk 1 above)
Comment (bfjelds) [reply SPECS/msft-golang/msft-golang.spec:13]: it seemed like the linter yelled about them being at the top.  i put them where the linter looked like it wanted.  i'll look to see if i can run the linter locally.

Comment (mfrw) [reply SPECS/msft-golang/msft-golang.spec:13]: The linter expects us to keep them where they currently are.
But for some reason (convention or whatever - I am not sure) we tend to keep them at the top.
/cc @PawelWMS

Comment (bfjelds) [reply SPECS/msft-golang/msft-golang.spec:13]: sorry, i misread your message.  i will move the macros back to the top.

Comment (bfjelds) [reply SPECS/msft-golang/msft-golang.spec]: moved macros back to the top

File: SPECS/msft-golang/msft-golang.spec (hunk 2 above)
Comment (mfrw) [reply SPECS/msft-golang/msft-golang.spec]: Sure, let me send the details offline.

**Review by mfrw — APPROVED** [review]
LGTM :)

**Review by rakshaa2000 — APPROVED** [review]

**Pull request merged**